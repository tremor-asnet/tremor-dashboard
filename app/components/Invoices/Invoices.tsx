"use client";

// Components
import InvoiceItem from "@/components/Invoices/InvoiceItem/InvoiceItem";

// Types
import { Invoice } from "@/types";

interface InvoicesProps {
  invoices: Invoice[];
}

const Invoices = ({ invoices }: InvoicesProps) => {
  const handleOnClick = () => {};

  const invoiceList = invoices.map(invoice => {
    const { date, code, price } = invoice;
    return (
      <InvoiceItem
        key={code}
        date={date}
        code={code}
        price={price}
        onClick={handleOnClick}
      />
    );
  });

  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md">
      <div className="text-[#344767] dark:text-white flex justify-between items-center">
        <h3 className="font-semibold">Invoices</h3>
        <button className="text-xs font-semibold px-[18px] py-[6px] border border-[#344767] dark:border-white rounded-md">
          VIEW ALL
        </button>
      </div>
      <ul className="flex flex-col gap-2 mt-6">{invoiceList}</ul>
    </div>
  );
};

export default Invoices;
