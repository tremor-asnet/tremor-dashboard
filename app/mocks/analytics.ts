export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
