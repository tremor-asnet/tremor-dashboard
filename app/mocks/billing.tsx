// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
