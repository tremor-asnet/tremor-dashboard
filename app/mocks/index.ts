export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
