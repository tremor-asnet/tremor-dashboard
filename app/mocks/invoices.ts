import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
