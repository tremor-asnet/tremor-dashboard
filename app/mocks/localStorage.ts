// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
