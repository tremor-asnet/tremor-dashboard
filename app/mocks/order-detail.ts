export const ORDER_DETAIL_DATA = [
  {
    id: 10421,
    order_code: "KF332",
    order_deliver_price: 14,
    order_tax: "1.95",
    created_at: "1 Nov, 10:20 AM",
    products: [
      {
        id: 1,
        name: "Gold Glasses",
        count: 1,
        price: 40,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/product-12.0b55635d.jpg",
      },
    ],
    track_order_info: {
      generate_order_id: "#1832412",
      generate_order_at: "1 Nov, 10:20 AM",
      transmited_to_courier_at: "1 Nov, 11:20 AM",
      delivered_at: "1 Nov, 10:20 PM",
    },
    billing_info: {
      last4_bank: "1234",
      owner_name: "Oliver Liam",
      company_name: "Viking Burrito",
      email: "oliver@burrito.com",
      vat: "FRB1235476",
    },
  },
];
