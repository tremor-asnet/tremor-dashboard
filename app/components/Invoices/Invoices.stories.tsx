// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import Invoices from "./Invoices";

// Mocks
import { MOCK_INVOICES } from "@/mocks/invoices";

const meta = {
  title: "Components/Invoices",
  tags: ["autodocs"],
  component: Invoices,
} as Meta<typeof Invoices>;

export default meta;

type Story = StoryObj<typeof meta>;

export const InvoiceDefault: Story = {
  args: {
    invoices: MOCK_INVOICES,
  },
};
