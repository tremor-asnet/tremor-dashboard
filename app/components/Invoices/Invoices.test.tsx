// Libs
import { render } from "@testing-library/react";

// Components
import Invoices from "@/components/Invoices/Invoices";

// Mocks
import { MOCK_INVOICES } from "@/mocks/invoices";

describe("Invoices Testing", () => {
  it("should match snapshot", () => {
    const { container } = render(<Invoices invoices={MOCK_INVOICES} />);

    expect(container).toMatchSnapshot();
  });
});
