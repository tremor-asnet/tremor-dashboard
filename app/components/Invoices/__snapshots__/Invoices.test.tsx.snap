// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoices Testing should match snapshot 1`] = `
<div>
  <div
    class="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md"
  >
    <div
      class="text-[#344767] dark:text-white flex justify-between items-center"
    >
      <h3
        class="font-semibold"
      >
        Invoices
      </h3>
      <button
        class="text-xs font-semibold px-[18px] py-[6px] border border-[#344767] dark:border-white rounded-md"
      >
        VIEW ALL
      </button>
    </div>
    <ul
      class="flex flex-col gap-2 mt-6"
    >
      <li
        class="w-full flex justify-between p-2"
      >
        <div>
          <p
            class="text-sm font-semibold text-[#344767] dark:text-white"
          >
            March, 21, 2021
          </p>
          <p
            class="text-xs mt-1 text-[#7b809a]"
          >
            #
            MS-123456
          </p>
        </div>
        <div
          class="flex gap-6 items-center"
        >
          <p
            class="text-sm text-[#7b809a]"
          >
            $
            180
          </p>
          <button
            class="flex gap-1 items-center text-[#344767] dark:text-white font-semibold"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
              />
            </svg>
            PDF
          </button>
        </div>
      </li>
      <li
        class="w-full flex justify-between p-2"
      >
        <div>
          <p
            class="text-sm font-semibold text-[#344767] dark:text-white"
          >
            Apr, 22, 2022
          </p>
          <p
            class="text-xs mt-1 text-[#7b809a]"
          >
            #
            MS-123457
          </p>
        </div>
        <div
          class="flex gap-6 items-center"
        >
          <p
            class="text-sm text-[#7b809a]"
          >
            $
            180
          </p>
          <button
            class="flex gap-1 items-center text-[#344767] dark:text-white font-semibold"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
              />
            </svg>
            PDF
          </button>
        </div>
      </li>
      <li
        class="w-full flex justify-between p-2"
      >
        <div>
          <p
            class="text-sm font-semibold text-[#344767] dark:text-white"
          >
            May, 23, 2023
          </p>
          <p
            class="text-xs mt-1 text-[#7b809a]"
          >
            #
            MS-999999
          </p>
        </div>
        <div
          class="flex gap-6 items-center"
        >
          <p
            class="text-sm text-[#7b809a]"
          >
            $
            180
          </p>
          <button
            class="flex gap-1 items-center text-[#344767] dark:text-white font-semibold"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
              />
            </svg>
            PDF
          </button>
        </div>
      </li>
      <li
        class="w-full flex justify-between p-2"
      >
        <div>
          <p
            class="text-sm font-semibold text-[#344767] dark:text-white"
          >
            Jun, 24, 2024
          </p>
          <p
            class="text-xs mt-1 text-[#7b809a]"
          >
            #
            MS-888888
          </p>
        </div>
        <div
          class="flex gap-6 items-center"
        >
          <p
            class="text-sm text-[#7b809a]"
          >
            $
            180
          </p>
          <button
            class="flex gap-1 items-center text-[#344767] dark:text-white font-semibold"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
              />
            </svg>
            PDF
          </button>
        </div>
      </li>
      <li
        class="w-full flex justify-between p-2"
      >
        <div>
          <p
            class="text-sm font-semibold text-[#344767] dark:text-white"
          >
            Jul, 25, 2025
          </p>
          <p
            class="text-xs mt-1 text-[#7b809a]"
          >
            #
            MS-476543
          </p>
        </div>
        <div
          class="flex gap-6 items-center"
        >
          <p
            class="text-sm text-[#7b809a]"
          >
            $
            180
          </p>
          <button
            class="flex gap-1 items-center text-[#344767] dark:text-white font-semibold"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5v1zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5v3zm4-3H19v1h1.5V11H19v2h-1.5V7h3v1.5zM9 9.5h1v-1H9v1zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm10 5.5h1v-3h-1v3z"
              />
            </svg>
            PDF
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
